run_wml_tests

```diff
@@ -155,70 +182,71 @@ class WesnothRunner:
-                print(repr(args))
+        if self.verbose >= Verbosity.SCRIPT_DEBUGGING:
+            print(repr(args))
+
         try:
             res = run_with_rerun_for_sdl_video(args, timeout)
         except subprocess.TimeoutExpired as t:
-            if self.verbose > 0:
-                print("Timed out (killed by Python timeout implementation)")
+            if expected_result != UnitTestResult.TIMEOUT:
+                print("Timed out (killed by Python timeout implementation), while running batch ({names})".format(
+                    names=", ".join([test.name for test in test_list])))
             res = subprocess.CompletedProcess(args, UnitTestResult.TIMEOUT.value, t.output or b'')
-        if self.verbose > 0:
+        if self.verbose >= Verbosity.OUTPUT_OF_PASSING_TESTS:
             print(res.stdout.decode('utf-8'))
-            if self.verbose > 1:
-                print("Result:", res.returncode)
-        num_passed = 1
-        if test_list[0].status == UnitTestResult.PASS:
-            num_passed = res.stdout.count(b"PASS TEST")
-            test_summary.pass_test(num_passed)
+            print("Result:", res.returncode)
         if res.returncode < 0:
             print("Wesnoth exited because of signal", -res.returncode)
             if options.backtrace:
                 print("Launching GDB for a backtrace...")
                 gdb_args = ["gdb", "-q", "-batch", "-ex", "start", "-ex", "continue", "-ex", "bt", "-ex", "quit", "--args"]
                 gdb_args.extend(args)
                 subprocess.run(gdb_args, timeout=240)
-            test_summary.crash_test()
-            test_summary.skip_test(len(test_list) - num_passed - 1)
+            test_summary.crash_test(test_list)
             raise UnexpectedTestStatusException()
         returned_result = UnitTestResult(res.returncode)
-        if returned_result != expected_result:
-            if self.verbose == 0:
-                for line in res.stdout.decode('utf-8').splitlines():
-                    if expected_result == UnitTestResult.PASS and line.startswith("PASS TEST"):
-                        continue
-                    print(line)
-            print("Failure, Wesnoth returned", returned_result, "for", test_list[num_passed].name, "but we expected", expected_result)
-            test_summary.fail_test()
-            test_summary.skip_test(len(test_list) - num_passed - 1)
+        if returned_result == expected_result:
+            test_summary.pass_test(test_list)
+        else:
+            if self.verbose < Verbosity.OUTPUT_OF_PASSING_TESTS:
+                # Batch mode only supports tests that should UnitTestResult.PASS, so the output
+                # here is likely to have many 'PASS TEST' lines with the failing test last.
+                # If support for batching other UnitTestResults is added then this needs to filter
+                # the output from the other tests.
+                print(res.stdout.decode('utf-8'))
```

Comment:
I'm not too happy that the filtering functionality was removed… the flood of PASS TEST messages when one test fails isn't great.

Reply:
If you can work out a reliable way to re-add it then please do, but let's make this script reliable before worrying about a flood of messages that say "PASS". Admittedly I'd take the opposite position if they said "FAIL", and that's just about what I expect a developer to be able to mentally filter out quickly.

Reply:
I'll think about it, then; for now we can leave it like this, I guess.